torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -157,15 +234,12 @@ std::vector<at::Tensor> ShardingUtil::ShardTensor(
   if (sharding.type() == xla::OpSharding::REPLICATED) {
     std::fill_n(shards.begin(), shards.size(), tensor);
   } else if (sharding.type() == xla::OpSharding::OTHER) {
-    XLA_CHECK_EQ(devices.size(), sharding.tile_assignment_devices().size())
+    CHECK_EQ(devices.size(), sharding.tile_assignment_devices().size())
         << "Invalid sharding tile_assignment_devices.size(): expected "
         << devices.size() << ", actual "
         << sharding.tile_assignment_devices().size();
-    // TODO(yeounoh) support higher-order topology
-    XLA_CHECK(sharding.tile_shape().dimensions_size() <= 2);
-    XLA_CHECK(tensor.sizes().size() >= sharding.tile_shape().dimensions_size())
-        << "Data dimensions should be at least as large as the tile shape "
-           "dimensions.";
+    CHECK(sharding.tile_shape().dimensions_size() <= 2);
+    CHECK(tensor.sizes().size() >= sharding.tile_shape().dimensions_size());
 
     auto tile_shape = sharding.tile_assignment_dimensions();
     for (int64_t core : sharding.tile_assignment_devices()) {
```

Comment:
let's not use core here, on v4 going forward, we use chip. I think it is better to just use `device` here to make it general.

Reply:
Good point. Ack